src/zope/container/directory.py

```diff
@@ -70,6 +70,7 @@ def __init__(self, context):
         pass
 
     def __call__(self, name):
+        from zope.container.folder import Folder
```

Comment:
Imports in non-test code should be at module scope.  If there is some kind of cycle, we should first try hard to break it another way:  if that is impossible, then the hidden import needs a comment highlighting the cycle.